### ISSUE TITLE
Update `ValidatorRecord` as per new spec

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#81]
## Description

`ValidatorRecord` has been outdated. Update it and all dependencies.

[validator_induction](https://github.com/sigp/lighthouse/tree/master/beacon_chain/validator_induction) is a definite dependency and it also needs to be updated to reflect the [new spec definition](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#routine-for-activating-a-validator). To do this, you'll need the presently-undefined `Deposit` and `DepositParameters` structs.

## Tasks

- Update the `ValidatorRecord` struct.
- Add the `Deposit` and `DepositParameters` structs.
- Remove the `types::validator_registration::ValidatorRegistration` struct.
- Update the [validator_induction](https://github.com/sigp/lighthouse/tree/master/beacon_chain/validator_induction) component.
- Update any other components that are affected (feel free to ping me if you have any questions around this, it might get tricky).

**Comment by paulhauner** [comment]
You're good to go on this one @g-r-a-n-t :)

**Comment by paulhauner** [comment]
Just letting you know that I updated this issue description and increased the scope (sorry!).

**Comment by paulhauner** [comment]
Following up from our discussion on Gitter, I think we should ditch the `ValidatorInductor` struct and just go with a simple function, much like the `process_deposit()` fn in the spec.

I think the function should have the following signature:

```rust
pub fn process_deposit(
    state: &mut BeaconState,
    deposit: &Deposit
    spec: &ChainSpec)
-> Result<usize, Error> 
```

Where the returned `usize` is the index of the new/updated validator in `BeaconState.validator_registry`.

The `Deposit` struct is not presently in the code base, however it is included in a PR I'm working on presently. I _should_ be able to get that in this afternoon, but the Rust meetup is this evening so it'll be tomorrow morning at the latest.

It's likely that I missed something in this fn signature, if something seems wrong it probably is :)

I hope this helps. As always, don't hesitate to ask questions :)

**Comment by g-r-a-n-t** [comment]
Hey @paulhauner, the `validator_change` stuff looks like it's pretty out of date. Is this worth refactoring or should I just remove it?

**Comment by paulhauner** [comment]
Just pull it out. I've been doing a lot of that lately :(

**Comment by paulhauner** [comment]
Resolved in #116
